### PR TITLE
Dashboard Prettified

### PR DESCRIPTION
**Pull request by vipj**
- Prettified Dashboard page and added DynamoDB functionality
- Fixed small issue with login function
- Added forgot password flow back to codebase

### CODE REVIEW

File: src/components/CompleteApplicationButton/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      );
+    } else if (this.state.status === "enabled") {
+      return (
+        <Button className={styles.button} href={"/application"}>
+          {this.state.text}
+        </Button>
+      );
+    } else {
+      return null;
+    }
```
Comment (kyleyarwood) [inline src/components/CompleteApplicationButton/index.js:31]: Can the status of this ever be anything but "enabled" or "disabled"? If not, I'm wondering if we'd be able to get rid of this case and make the above one an else.

File: src/components/DashboardSidebarButtons/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import Button from "react-bootstrap/Button";
+
+import styles from "./styles.module.css";
+
+export default class DashboardSidebarButtons extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      text: props.text,
```
Comment (kyleyarwood) [inline src/components/DashboardSidebarButtons/index.js:11]: Is this property needed for this button? I don't see it used anywhere

File: src/views/DashboardPage/index.js [hunk 3]
```diff
@@ -6,41 +6,99 @@ import Button from "react-bootstrap/Button";
...
-          return;
-        }
-      });
-      isLoggedIn = true;
-    });
-  } else {
-    document.getElementByIdById("display_error").innerHTML = "Not Logged In";
-  }
-};
-
```
Comment (kyleyarwood) [inline src/views/DashboardPage/index.js:37]: I'm not super well-versed in React haha, so I'm kinda curious, what was this doing and why were we able to get rid of it?

File: src/components/DashboardSidebarButtons/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      );
+    } else if (this.state.type === "application") {
+      return (
+        <Button href={"/application"} variant="link" className={styles.button} block>
+          <h1 className={styles.text}>
+            Application
+          </h1>
+        </Button>
+      );
+    } else if (this.state.type === "logout") {
```
Comment (dunja-tomic) [inline src/components/DashboardSidebarButtons/index.js:33]: This code block should call the logout function!

File: src/views/ForgotPasswordPageChange/index.js [hunk 5]
```diff
@@ -1,34 +1,76 @@
...
+      });
+    } else {
+      this.setState({ err: "noMatch" });
+    }  
   };
+  
 
   displayErrors = () => {
-    if (this.state.err) {
+    if (this.state.err === "noCode") {
```
Comment (dunja-tomic) [inline src/views/ForgotPasswordPageChange/index.js:67]: I like this, nice!

File: src/components/DashboardSidebarButtons/index.js (hunk 2 above)
Comment (vipj) [reply src/components/DashboardSidebarButtons/index.js:11]: Oops. You're right

File: src/components/CompleteApplicationButton/index.js (hunk 1 above)
Comment (vipj) [reply src/components/CompleteApplicationButton/index.js:31]: I had that for when I was testing but forgot to take it out - I'll do that now

File: src/components/DashboardSidebarButtons/index.js (hunk 4 above)
Comment (vipj) [reply src/components/DashboardSidebarButtons/index.js:33]: Right, on it

File: src/views/DashboardPage/index.js (hunk 3 above)
Comment (vipj) [reply src/views/DashboardPage/index.js:37]: It was using the window.onLoad() function. Dunja wanted to use something from the React component lifecycle, so I just moved it to the constructor (which is below)

File: src/views/ForgotPasswordPageChange/index.js (hunk 5 above)
Comment (vipj) [reply src/views/ForgotPasswordPageChange/index.js:67]: Thanks :)

**Comment by vipj** [comment]
- Added logout function to logout button on dashboard
- Made the dashboard send you to /login if you're not already logged in

File: src/views/DashboardPage/index.js (hunk 3 above)
Comment (dunja-tomic) [reply src/views/DashboardPage/index.js:37]: Yeah they accomplish the same thing, but this was is just more react-y

**Pull request merged**